rocks-bin/src/project/write_new.rs

```diff
@@ -0,0 +1,301 @@
+use std::{error::Error, path::PathBuf, str::FromStr};
+
+use clap::Args;
+use eyre::{eyre, Result};
+use inquire::{
+    ui::{RenderConfig, Styled},
+    validator::Validation,
+    Confirm, Select, Text,
+};
+use itertools::Itertools;
+use spdx::LicenseId;
+use spinners::{Spinner, Spinners};
+
+use crate::utils::github_metadata::{self, RepoMetadata};
+use rocks_lib::{project::Project, rockspec::LuaDependency};
+
+// General notes and ideas:
+// - Create a subdirectory for the project, unless a path is explicitly passed.
+// - Automatically detect build type by inspecting the current repo (is there a Cargo.toml? is
+//   there something that tells us it's a lua project?).
+
+#[derive(Args)]
+pub struct NewProject {
+    /// The directory of the project.
+    #[arg(long)]
+    directory: Option<PathBuf>,
+
+    /// The project's name.
+    #[arg(long)]
+    name: Option<String>,
+
+    /// The description of the project.
+    #[arg(long)]
+    description: Option<String>,
+
+    /// The license of the project. Generic license names will be inferred.
+    #[arg(long, value_parser = clap_parse_license)]
```

Comment:
question: Does this fail if the license can't be parsed?

Which can be the case if

- the project has an esoteric/custom license 
- there's no license 
- there are multiple licenses 

If so, should we just fall back to `UNKNOWN` and print a warning?

Reply:
I took the KISS approach and allowed for the following:
- SPDX-recognized license
- `none` for no license

When no license is provided, the `project.rockspec` has an `*** enter license ***` field instead. I don't believe the rockspec format even allows for multiple licenses, but do correct me if I'm wrong. Implementing that shouldn't take long :)

Reply:
rockspecs don't allow for multiple licenses, but I've encountered projects where automatic license detection libraries failed because they detected more than one license. 
I presume this command would fail in such a case?

Reply:
The way it's structured it won't fail when there's more than one license. I believe it'll just pick the first one it finds. We're retrieving this data from github and we're receiving the data in a struct from a crate so it's impossible for things to go haywire with variable length data :)